Source/Scripts/Syntax.cs-schema

```diff
@@ -50,6 +63,14 @@ abstract class Type : NodeWithOrigin
 {
 }
 
+abstract class TypeProxy : Type
+{
+}
+
+class InferredTypeProxy : TypeProxy
```

Comment:
Why is this type here? This feels like something that's part of resolution, not the grammar.

Reply:
I agree, and I've refactored the parser and node logic so that we can faithfully serialize/deserialize ASTs without needing these types.